store/store.go

```diff
@@ -403,7 +395,7 @@ func (s *Store[H]) flushLoop() {
 		// and notify waiters if any + increase current read head height
 		// it is important to do Pub after updating pending
 		// so pending is consistent with atomic Height counter on the heightSub
-		s.heightSub.Pub(headers...)
+		s.heightSub.Pub(headers...) // TODO(oleg): the only place where we mutate heightSub.
```

Comment:
Literally what comments says. Well, we also have `s.heightSub.Sub` where we add new subscribers but headers are updated only here.